### ISSUE TITLE
False positive in empty_enum_arguments rule

### ISSUE THREAD

**Issue by gjeck** [realm/SwiftLint#2773]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`enum` cases with associated values with no label cause a false positive on the `empty_enum_arguments` rule.

### Environment

* SwiftLint version: `0.32.0`
* Installation method used: Homebrew/CocoaPods
* Paste your configuration file: n/a
* Xcode version: `10.2.1`
* Swift version: `5`

```swift
// Given:
enum Event {
  case started(String)
  case finished(String)
}

// This triggers a violation:
let event = Event.started("cool")
switch event {
case .started(_): // Empty Enum Arguments Violation
  print("started")
case .finished(_): // Empty Enum Arguments Violation
  print("finished")
}
```

![Image of violation](https://i.imgur.com/Hog7CQg.png)

**Comment by marcelofabri** [comment]
That's not a false positive, the rule wants you to write this:

```swift
switch event {
case .started:
  print("started")
case .finished:
  print("finished")
}
```

**Comment by gjeck** [comment]
OMG sorry!

**Comment by Jiropole** [comment]
I am getting this warning in strange places that have nothing to do with enums. For example, I have 

- `model` - an `ObservableObject`
- `email` - a string
- `isValidEmail()` a private extension on String

Under these conditions, with code
```
if !model.email.isValidEmail() { ... }
```
SwiftLint complains `empty_enum_arguments`, for reasons unknown.

**Comment by marcelofabri** [comment]
@Jiropole can you make sure you're using the latest SwiftLint version?

**Comment by Jiropole** [comment]
Thanks @marcelofabri 👍. Quite right; fixed sometime since 0.43.1.
